drivers/sensors/spacemouse_module.c

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "spacemouse_module.h"
+#include "pointing_device_internal.h"
+#include "uart.h"
+
+// datasheet available at https://3dconnexion.com/cn/wp-content/uploads/2020/02/HW-Spec-3DX-700039_Rev001_serial.pdf
+
+#define SPACEMOUSE_INPUT_OFFSET (8192)
+
+bool spacemouse_send_command(uint8_t cmd) {
+    uart_write(cmd);
+    uint8_t buf[2];
+    uart_receive(buf, 2);
+    return (buf[0] == cmd && buf[1] == SPACEMOUSE_CMD_END);
+}
+
+/**
+ * @brief Set the zero position of the module
+ *
+ * @return true command ran successfully
+ * @return false command failed
+ */
+bool spacemouse_cmd_set_zero_position(void) {
+    return spacemouse_send_command(SPACEMOUSE_CMD_SET_ZERO_POSITION);
+}
+
+/**
+ * @brief Starts automatic transmission of data, at 30ms invervals
+ * Automatic data transmission happens at 30 ms intervals, but device can be polled at 100/s or 10ms intervals
+ * Since 10ms is what pointing device polling defaults to, we don't need need the stream command, but
+ * it is here for completeness, in case somebody wants to implement it elsewhere.
```

Comment:
There's no default pointing task throttle, a 1ms throttle exists when using split pointing and some of pointing device drivers specify a default throttle.

Reply:
yeah, wasn't really sure about that.  Though, it doesn't appear to have any issues with polling more frequently (eg, my testing didn't have it set, and it seemed perfectly stable) .

Also, not sure the best way to handle forcibly throttling it.

Reply:
I guess how the cirque or azoteq does it